swap_meet/vendor.py

```diff
@@ -1,2 +1,73 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        filtered_items = []
+        for item in self.inventory:
+            if item.category == category:
+                filtered_items.append(item)
+        return filtered_items
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item in self.inventory:
+            if their_item in vendor.inventory:
+                self.remove(my_item)
+                self.add(their_item)
+                vendor.remove(their_item)
+                vendor.add(my_item)
+                return True
+            else:
+                return False
+        else:
+            return False
```

Comment:
We could simplify the if-statements a bit if we do our valid data check at the top of the function as a guard statement:
```suggestion
        if my_item not in self.inventory or their_item not in friend.inventory:
            return False
            
        self.remove(my_item)
        self.add(their_item)
        vendor.remove(their_item)
        vendor.add(my_item)
        return True
```